tests/test_pvc_concurrent_deletion_creation.py

```diff
@@ -0,0 +1,223 @@
+"""
+Test to verify concurrent creation and deletion of multiple PVCs
+"""
+import os
+import subprocess
+import pytest
+import yaml
+import logging
+from munch import munchify
+import ocs.defaults as defaults
+
+from ocs import ocp
+from ocs import exceptions
+from ocsci import tier1, ManageTest
+from utility import templating
+
+log = logging.getLogger(__name__)
+TEMPLATE_DIR = "templates/ocs-deployment"
+TEMP_YAML_FILE_SC = '/tmp/temp_file_sc.yaml'
+TEMP_YAML_FILE_PVC = '/tmp/temp_file_pvc.yaml'
+SC = ocp.OCP(kind='StorageClass', namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+PVC = ocp.OCP(kind='PersistentVolumeClaim',
+              namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+PV = ocp.OCP(kind='PersistentVolume',
+             namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+
+
+@pytest.fixture(scope='class')
+def test_fixture(request):
+    """
+    Removing the resources created for this test
+    """
+    def finalizer():
+        teardown()
+    request.addfinalizer(finalizer)
+
+
+def teardown():
+    """
+    Delete storage class
+    Delete temporary yaml files
+    """
+    assert SC.delete(yaml_file=TEMP_YAML_FILE_SC)
+    os.remove(TEMP_YAML_FILE_PVC)
+    os.remove(TEMP_YAML_FILE_SC)
+
+
+def create_storage_class(**kwargs):
+    """
+    Create storage class
+
+    Args:
+        **kwargs: key, value pairs for yaml file substitution
+    """
+    sc_yaml = os.path.join(TEMPLATE_DIR, "StorageClass.yaml")
+    sc_name = kwargs['sc_name']
+
+    sc_yaml_file = templating.generate_yaml_from_jinja2_template_with_data(
+        sc_yaml, **kwargs
+    )
+    with open(TEMP_YAML_FILE_SC, 'w') as yaml_file:
+        yaml.dump(sc_yaml_file, yaml_file, default_flow_style=False)
+    log.info(f'Creating Storage Class {sc_name}')
+    assert SC.create(yaml_file=TEMP_YAML_FILE_SC)
+    log.info(f'Created Storage Class {sc_name}')
+
+
+def create_multiple_pvc(pvc_base_name, number_of_pvc):
+    """
+    Create PVCs
+
+    Args:
+        pvc_base_name (str): Prefix of PVC name
+        number_of_pvc (int): Number of PVCs to be created
+    """
+    pvc_yaml = os.path.join(TEMPLATE_DIR, "PersistentVolumeClaim.yaml")
+    pvc_parms = {'sc_name': 'test-sc', 'pvc_name': pvc_base_name}
+
+    pvc_yaml_file = templating.generate_yaml_from_jinja2_template_with_data(
+        pvc_yaml, **pvc_parms
+    )
+    pvc_obj = munchify(pvc_yaml_file)
+
+    for count in range(1, number_of_pvc + 1):
```

Comment:
@jilju are we creating a separate project/namespace for new pvcs. Its always better to have a new project rather than using default/openshift-storage.